### ISSUE TITLE
`no-cycle` does not raise error on circular  dependency?

### ISSUE THREAD

**Issue by Yaojian** [import-js/eslint-plugin-import#2004]
There are three files:

`src/a.ts`:
```ts
import y from "./b";

const x: number = y + 1;
export default x;
```

`src/b.ts`:
```ts
import x from "./a";

const y: number = x + 1;
export default y;
```

`src/c.ts`:
```ts
import x from "./a";
import y from "./b";

console.log(x, y);
```

and the configuration is:
```json
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript"
  ],
  "plugins": [
    "@typescript-eslint",
    "import"
  ],
  "parser": "@typescript-eslint/parser",
  "settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts",
        ".tsx"
      ]
    }
  },
  "rules": {
    "import/no-cycle": [
      "error",
      {
        "maxDepth": 10,
        "ignoreExternal": true
      }
    ]
  }
}
```

Given the command
```
$ yarn eslint src\c.ts
yarn run v1.22.10
$ D:\Projects\graphact\server\node_modules\.bin\eslint src\c.ts
Done in 2.03s.
```
**no** error occurs while `c.ts` imports both `a.ts` and `b.ts`, which are circular importing each other!

Is it the desired behavior, or I missed something?

**Comment by ljharb** [comment]
You may also need to set the import/extensions setting, since these aren’t js files.

**Comment by Yaojian** [comment]
Still not working after adding `import/extensions`. 

I setup a test project on  https://github.com/Yaojian/no-cycle-test

**Comment by DerGernTod** [comment]
i have the very same issue with a similar setup

**Comment by Smrtnyk** [comment]
I have the same issue and it would help a lot having this working :)

**Comment by Slapbox** [comment]
Check out #2103.

The next step here I think is to install `eslint-plugin-import` via setting a specific commit ID in the package.json and to test if we can find a commit where either of these bugs were not present. There's not many commits, so most of the work involved here (should) be just starting/stopping costs of taking on a next task and waiting for npm installs.

**Comment by soryy708** [comment]
I don't expect `src/c.ts` to have any warnings about cycles, because it isn't involved in cycles.
I expect, as aligned with existing behavior, to get a report on an import if it's involved in a cycle with the currently linted file.
So I expect to have it reported in `src/a.ts` and in `src/b.ts`.

**Comment by ljharb** [comment]
Thanks, not sure why I didn't catch that originally. c has no cycle; a and b do.
